server/src/main/java/org/opensearch/action/search/SearchRequest.java

```diff
@@ -102,6 +102,8 @@ public class SearchRequest extends ActionRequest implements IndicesRequest.Repla
 
     private Boolean allowPartialSearchResults;
 
+    private Boolean ignoreUnavailableShards;
```

Comment:
What are all the possible interactions between `allowPartialSearchResults` and `ignoreUnavailableShards`?

Would another option be to differentiate between `allowPartialSearchResults` is `null` (which currently defaults to `true`) versus the case where it's explicitly passed as `true`?

Reply:
> What are all the possible interactions between allowPartialSearchResults and ignoreUnavailableShards?

Essentially, `allowPartialSearchResults` (ignore any type of shard failures) option supersedes `ignoreUnavailableShards` (only ignores unavailable shards). But given that the `allowPartialSearchResults` is not available with the `_delete_by_query`, `_update_by_query` and `_reindex`, it makes sense to expose more limited `ignoreUnavailableShards` instead of changing existing behavior of `allowPartialSearchResults`.

> Would another option be to differentiate between allowPartialSearchResults is null (which currently defaults to true) versus the case where it's explicitly passed as true?

I am not sure about that option, since it changes the existing behavior and adds 3 cases for boolean value

Reply:
Interestingly, the `ignore_unavailable` parameter is already part of search API documentation - https://opensearch.org/docs/latest/api-reference/search/